src/Uno.UI.Composition/Composition/ContainerVisual.skia.cs

```diff
@@ -9,6 +12,19 @@ public partial class ContainerVisual : Visual
 	private List<Visual>? _childrenInRenderOrder;
 	private bool _hasCustomRenderOrder;
 
+	private (Rect rect, bool isAncestorClip)? _clipRectFromArrangeCycle;
+
+	/// <summary>
+	/// Layout clipping is usually applied in the element's coordinate space.
+	/// However, for Panels and ScrollViewer headers specifically, WinUI applies clipping in the parent's coordinate space.
+	/// So, isAncestorClip will be set to true for Panels and ScrollViewer headers, indicating that clipping is in parent's coordinate space.
+	/// </summary>
+	internal (Rect rect, bool isAncestorClip)? ClipRectFromArrangeCycle
```

Comment:
A shorter name could be `LayoutClipRect` which also aligns with the terminology from WinUI sources.